presto-thrift-api/src/main/java/com/facebook/presto/thrift/api/udf/ThriftUdfPage.java

```diff
@@ -54,9 +52,9 @@ public ThriftUdfPageFormat getPageFormat()
 
     @Nullable
     @ThriftField(value = 2, requiredness = OPTIONAL)
```

Comment:
I think we should model this as a ThriftUnion, since it is an either/or with ThriftSerializedPage vs PrestoThriftPage. Clients won't have to write guard code to check to make sure that one of these is non-empty, and it would clarify the API.

Reply:
The problem is that we Java doesn't have that so there's not really a natural way to do it. The Thrift connector API also just used this model, which is why I did this too. But I agree with you on principal.